packages/delegation-revoke-archiver/src/delegationRevokeArchiverConsumerServiceV2.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  CorrelationId,
+  DelegationEventEnvelopeV2,
+  generateId,
+  missingKafkaMessageDataError,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { match } from "ts-pattern";
+import {
+  // getInteropHeaders,
+  logger,
+  RefreshableInteropToken,
+} from "pagopa-interop-commons";
+// import { getInteropBeClients } from "./clients/clientsProvider.js";
+
+// const { agreementProcessClient, purposeProcessClient } = getInteropBeClients();
+
+export async function handleMessageV2({
+  decodedKafkaMessage,
+  // refreshableToken,
+  partition,
+  offset,
+}: {
+  decodedKafkaMessage: DelegationEventEnvelopeV2;
+  refreshableToken: RefreshableInteropToken;
+  partition: number;
+  offset: string;
+}): Promise<void> {
+  await match(decodedKafkaMessage)
+    .with({ type: "ConsumerDelegationRevoked" }, async (delegationMsg) => {
+      const correlationId = delegationMsg.correlation_id
+        ? unsafeBrandId<CorrelationId>(delegationMsg.correlation_id)
+        : generateId<CorrelationId>();
+
+      const loggerInstance = logger({
```

Comment:
I would move the logger setup in the `processMessage` main function in the index, so that we set up everything there, before routing from a message to its handler